### PR TITLE
ci: add nancy

### PR DESCRIPTION
**Pull request by jeinfeldt**
Hello,

we would benefit from using [Nancy](https://github.com/sonatype-nexus-community/nancy) to check our dependencies.
@tessig opened a PR for this previously https://github.com/i-love-flamingo/flamingo/pull/69
before `go.mod` was fully supported.

Since flamingo does not use travis anymore for CI, lets try with github actions.
Workflow suggested by: https://github.com/marketplace/actions/nancy-for-github-actions

For now I would suggest to only do this on PR, as we would otherwise break the CI on master.
After all vulnerabilities are fixed, we can adjust the behaviour.
Of course we can discuss how to integrate this workflow / naming things.

The current vulnerability is related to
https://github.com/spf13/viper/pull/957

If we decide to merge this closes https://github.com/i-love-flamingo/flamingo/pull/69

### CODE REVIEW

**Comment by carstendietrich** [comment]
We could move this to the `Static checks` part, and we should have a look at https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events to run the dependency check on a daily basis..

**Comment by carstendietrich** [comment]
@jeinfeldt please check the scheduled trigger and try to do a go get on etcd to override the vuln dependency

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -54,4 +54,12 @@ jobs:
       - name: Go Generate
         run: |
           go generate ./...
-          git diff --quiet || (echo 'generated go files are not up to date, check go generate, go.sum and go.mod' ; git diff ; exit 1)
\ No newline at end of file
+          git diff --quiet || (echo 'generated go files are not up to date, check go generate, go.sum and go.mod' ; git diff ; exit 1)
+      - name: Set up Go 1.x in order to write go.list file
+        uses: actions/setup-go@v2
+        with:
```
Comment (tessig) [inline .github/workflows/main.yml:60]: we don't need this, it's already above

File: .github/workflows/daily.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  vulnerability-scan:
+    name: Vulnerability scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      - name: Set up Go 1.x in order to write go.list file
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.13
```
Comment (tessig) [inline .github/workflows/daily.yml:17]: just take the most actual version via 1.*

**Review by tessig — CHANGES_REQUESTED** [review]

**Review by tessig — APPROVED** [review]
very nice, thanks!

**Pull request merged**